### PR TITLE
DestroyBullet

### PR DESCRIPTION
**Pull request by MrDuck557**
Blue's Message

I implemented it as the field `destroyBullet`

Demo video: 

https://user-images.githubusercontent.com/48808663/140625829-cd4f3ba8-8472-448f-bbec-394a31e659b0.mp4

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by MrDuck557** [comment]
Yes the bullets don't affect derelict

**Comment by BlueWolf3682** [comment]
Rotation of 0 is preferable - if I need a random rotation, I can use a frag bullet.

File: core/src/mindustry/entities/comp/BuildingComp.java [hunk 1]
```diff
@@ -1032,7 +1032,12 @@ public void tapped(){
 
     /** Called *after* the tile has been removed. */
     public void afterDestroyed(){
-
+        if(block.destroyBullet != null){
+            /** I really do not like that the bullet will not destroy derelict
```
[the remark this answers is not in the document]
Comment (MrDuck557) [reply core/src/mindustry/entities/comp/BuildingComp.java]: Fixed

File: core/src/mindustry/entities/comp/BuildingComp.java [hunk 2]
```diff
@@ -1032,7 +1032,12 @@ public void tapped(){
...
+             but I can't do anything about it without using a random team
+             which may or may not cause issues with servers and js*/
+            block.destroyBullet.create(this, Team.derelict, x, y, rotdeg());
```
[the remark this answers is not in the document]
Comment (MrDuck557) [reply core/src/mindustry/entities/comp/BuildingComp.java]: Fixed

**Pull request merged**